api/src/pcapi/routes/pro/collective_offers.py

```diff
@@ -207,3 +207,31 @@ def create_collective_offer_template_from_collective_offer(
         )
 
     return collective_offers_serialize.CollectiveOfferTemplateResponseIdModel.from_orm(collective_offer_template)
+
+
+@private_api.route("/collective/offers-template/<offer_id>", methods=["PATCH"])
+@login_required
+@spectree_serialize(
+    response_model=collective_offers_serialize.GetCollectiveOfferTemplateResponseModel,
+    api=blueprint.pro_private_schema,
+)
+def edit_collective_offer_template(
+    offer_id: str, body: collective_offers_serialize.PatchCollectiveOfferTemplateBodyModel
+) -> collective_offers_serialize.GetCollectiveOfferTemplateResponseModel:
+    dehumanized_id = dehumanize_or_raise(offer_id)
+    try:
+        offerer = offerers_api.get_offerer_by_collective_offer_template_id(dehumanized_id)
+    except offerers_exceptions.CannotFindOffererForOfferId:
+        raise ApiErrors({"offerer": ["Aucune structure trouvée à partir de cette offre"]}, status_code=404)
+    else:
+        check_user_has_access_to_offerer(current_user, offerer.id)
+
+    try:
+        offers_api.update_collective_offer(
```

Comment:
je me demande si ca vaudrait pas le coup de faire un `update_collective_offer_template` qui serait utilisé uniquement ici. Ca permettrait d'avoir une fonction plus clean que celle qui gère l'update dans le cas légacy/template ou pas template ...

Reply:
En fait ce serait un duplicat de l'autre. Les differences pour chacun des ces 4 cas sont vraiment minimes.

Reply:
pas vraiment d'accord on pourrait virer toute la partie legacy et ca simplifie le début aussi

Reply:
j'ai fait une solution intermediaire, dis moi si ça te convient. (j'en ai profité pour virer le if not offer_to_update qui ne peut jamais avoir lieu car la fonction apellante a besoin que l'offre existe pour faire le controle de droits)